src/hazmat/primecount.rs

```diff
@@ -0,0 +1,117 @@
+use crypto_bigint::{Uint, U256};
+
+/// Estimate the number of primes smaller than x using the asymptotic expansion of Li(x) with three terms. 𝜋(𝑥)/Li(x)
+/// converges to 1 from below, which means that the estimation of number of primes for the ranges relevant here is a
+/// slight overestimate.
+/// Uses the formula 𝜋(𝑥) ~ (x/ln(x)) * (1 + 1!/ln(x) + 2!/(ln(x)^2) + 3!/(ln(x)^3)).
+pub fn estimate_pi_x<const LIMBS: usize>(x: &Uint<LIMBS>) -> Uint<LIMBS> {
+    let ln_x = ln(x);
+    let term1 = x / Uint::from_u64(ln_x);
+    let ln_x = ln_x as f64;
+    let ln_x_sq = ln_x * ln_x;
+    let term2 = (1.0 + 1.0 / ln_x + 2.0 / (ln_x_sq) + 6.0 / (ln_x_sq * ln_x)) as u64;
+    term1 * Uint::<LIMBS>::from_u64(term2)
+}
+
+// Calculate the natural logarithm of a big integer using the relation ln(x) = log₂(x) / log₂(e).
+// Returns a truncated integer representation of ln(x), which means this is a "decent" approximation for large integers
+// in the range of 2^64 to 2^8192. Smaller integers will be less accurate.
+fn ln<const LIMBS: usize>(x: &Uint<LIMBS>) -> u64 {
```

Comment:
How much do we gain with `U256` compared to using `u64`? Specifically, is it noticeable when used in another approximation (`Li(x)`)? Also, it would be useful to specify exactly how big the error is.

Reply:
When used to approximate Li(x) there is not discernible difference at all between using my 128-bit LN2 and core::f64::consts::LN2. 
The errors from the [comment above](https://github.com/entropyxyz/crypto-primes/pull/81#discussion_r2014136862) are exactly the same.